### ISSUE TITLE
PRoblems with Polly

### ISSUE THREAD

**Issue by rebop** [zakius/Polly#11]
Hi folks. Used a previous release on Waterfox classic primarily for Github issues. It crashed my browser very badly creating 212 .tmp files and requiring killing processes.

I was asked to try the latest and I am right now. This does not seem to crash, but after opening a number of tabs (not yet calculated) the Polly dropdown is blank for EVERY page. Must close and reopen browser to see it again.

**Comment by zakius** [comment]
Once that happens can you go to `about:debugging`, enable addon debugging and then inspect Polly to see what pops in the console? There likely will be a lot of stuff printed by the browser itself but if anything helpful appears that will be in the last few lines and point out file name and line properly. And for the next version I'll add debugging information for Promise rejections

**Comment by rebop** [comment]
I will give it my best and see what I can find.

It does come back as I close tabs so must be memory related in some way.

**Comment by rebop** [comment]
Is this helpful at all? I know not following your instructions, but I was at the issue:

running injector  injector.js:19:1
injector received response  injector.js:2:5
injecting polyfills into document tree  injector.js:7:5
Array [  ]  injector.js:9:5
JQMIGRATE: Migrate is installed, version 1.4.1  jquery-migrate.min.js:2:542
Loading failed for the <script> with source “https://www.well.com/wp-content/plugins/pixelyoursite/js/public.js?ver=5.2.0”.  login.html:155
Loading failed for the <script> with source “https://stats.wp.com/e-201810.js”.  login.html:567
Empty string passed to getElementById().

**Comment by rebop** [comment]
Addon debugging was enabled. However when I click debug for Polly I get this:

An incoming request to permit remote debugging connection was detected. A remote client can take complete control over your browser!

Client Endpoint: 127.0.0.1:60946
Server Endpoint: 127.0.0.1:6080

Allow connection?

Having never seen that warning before I have to check.

**Comment by rebop** [comment]
And on Github where Polly works, but alos alternates with blank dropdown:

running injector  injector.js:19:1
injector received response  injector.js:2:5
was handled by response handler  injector.js:4:9
Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src https://github.githubassets.com”). Source: (function (NAVIGATOR, OBJECT) {

    OBJ....  github.com:1
ReferenceError: queueMicrotask is not defined[Learn More]

**Comment by zakius** [comment]
> Addon debugging was enabled. However when I click debug for Polly I get this:
> 
> An incoming request to permit remote debugging connection was detected. A remote client can take complete control over your browser!
> 
> Client Endpoint: 127.0.0.1:60946
> Server Endpoint: 127.0.0.1:6080
> 
> Allow connection?
> 
> Having never seen that warning before I have to check.

Yes, extension debugging is like that in older versions of Firefox (and derivatives).

When you allow it new window will pop up and there you'll see what happens in BackgroundScripts

**Comment by rebop** [comment]
Just an update: Clicking the icon repeatedly when it shows blank will populate the choices. FYI.
